rpc/src/error.rs

```diff
@@ -89,6 +89,15 @@ define_error! {
                 format_args!("method not found: {}", e.method)
             },
 
+        HttpRequestFailed
+            {
+                status: http::StatusCode,
+                body: String
+            }
+            | e | {
+                format_args!("HTTP request failed with non-200 status code: {}, body: {}", e.status, e.body)
```

Comment:
Is there ever a need to dump the whole response body (which could be arbitrarily large) in error handling? Please leave it to the logs.

Reply:
Good point!

Reply:
[`2dc92cc`](https://github.com/informalsystems/tendermint-rs/pull/1361/commits/2dc92cc35a6beb8afca2c9dd6dd99f8eee178ed0)